File: opendemo_output/nodejs/nodejs-nodejs-demo-7/example1.js
/**
 * example1.js - 对象解构基础与进阶用法
 * 展示如何从对象中提取属性，并使用默认值和重命名
 */

// 模拟一个用户对象
const user = {
  name: 'Alice',
  age: 25,
  city: '北京',
  email: 'alice@example.com'
};

// 基础解构：提取指定字段
const { name, age } = user;
console.log(`【示例1-基础】用户姓名：${name}，年龄：${age}`);

// 重命名字段：将 city 解构为 residence
const { city: residence } = user;
console.log(`【示例1-重命名】居住地：${residence}`);

// 设置默认值：防止 undefined
const { nickname = '未设置昵称' } = user;
console.log(`【示例1-默认值】昵称：${nickname}`);

// 混合使用：重命名 + 默认值
const { phone = '无联系方式', email: contact } = user;
console.log(`【示例1-混合】联系邮箱：${contact}，电话：${phone}`);

// 嵌套对象解构
const student = {
  personal: {
    name: 'Bob',
    grade: 'A'
  },
  school: '清华大学'
};

const {
  personal: { name: studentName, grade },
  school
} = student;

console.log(`【示例1-嵌套】学生 ${studentName} 成绩 ${grade}，就读于 ${school}`);

File: opendemo_output/python/libraries/numpy/logic-functions/code/example1.py
"""
文件: example1.py
功能: 演示基于用户名和角色的访问控制逻辑
场景: 系统权限验证
"""

def has_access(username, is_active, role):
    """
    判断用户是否有系统访问权限

    权限规则:
    - 用户必须处于激活状态 (is_active 为 True)
    - 并且必须是管理员(role=='admin') 或者 是特殊用户 'alice'

    参数:
        username (str): 用户名
        is_active (bool): 是否激活
        role (str): 用户角色

    返回:
        bool: 是否有访问权限
    """
    # 使用逻辑与(and)确保两个条件都满足
    # 使用逻辑或(or)实现多路径授权
    # 括号保证 or 表达式先计算
    return is_active and (role == "admin" or username == "alice")


# 测试用例
if __name__ == "__main__":
    print("=== 示例1：权限验证 ===")
    
    # 测试用户alice（即使不是admin也允许）
    result1 = has_access("alice", True, "user")
    print(f"用户 'alice' 权限检查结果: {result1}")
    
    # 测试普通用户bob（未被特别授权）
    result2 = has_access("bob", True, "user")
    print(f"用户 'bob' 权限检查结果: {result2}")